client/delegations_test.go

```diff
@@ -0,0 +1,408 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/theupdateframework/go-tuf/data"
+	"github.com/theupdateframework/go-tuf/verify"
+)
+
+var (
+	defaultPathPatterns = []string{"tmp", "*"}
+	noMatchPathPatterns = []string{"vars", "null"}
+)
+
+func TestDelegationsIterator(t *testing.T) {
```

Comment:
Let's add some hash prefix tests. I didn't get to writing these tests today, but I tested the client on an example delegated hashed-bins repo, and it worked properly.

Reply:
HashPrefix / path logic is handled by DelegatedRole in data/types_go and is already tested in data/types_test.go `TestDelegatedRolePathMatch`. We shouldn't test the internals of dependencies